### PR TITLE
Add parser

### PR DESCRIPTION
**Pull request by yosradirgham**

### CODE REVIEW

**Comment by PierrickV** [comment]
We should use this PR to fix:
- https://github.com/SekoiaLab/platform/issues/11932
- https://github.com/SekoiaLab/platform/issues/26009
- https://github.com/SekoiaLab/platform/issues/28115
- https://github.com/SekoiaLab/platform/issues/26479

File: Fortinet/fortigate/_meta/fields.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+tmp.icmp_code:
+  name: tmp.icmp_code
+  type: keyword
+  description: ""
+
+tmp.icmp_type:
+  name: tmp.icmp_type
+  type: keyword
+  description: ""
```
Comment (PierrickV) [inline Fortinet/fortigate/_meta/fields.yml:19]:
```suggestion
tmp.icmp_code:
  name: tmp.icmp_code
  type: keyword
  description: ""

tmp.icmp_type:
  name: tmp.icmp_type
  type: keyword
  description: ""
```

Could you please use [ECS ICMP fields](https://www.elastic.co/guide/en/beats/packetbeat/current/exported-fields-icmp.html)

File: Fortinet/fortigate/_meta/fields.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+fortigate.event.severity:
+  name: fortigate.event.severity
+  type: keyword
+  description: ""
+
+dns.rrtype:
```
Comment (PierrickV) [inline Fortinet/fortigate/_meta/fields.yml:26]: You can use [dns.question.type](https://www.elastic.co/guide/en/ecs/current/ecs-dns.html#field-dns-question-type)

File: Fortinet/fortigate/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+          source.address: "{{parsed_event.message.remip or parsed_event.message.srcip or parsed_event.message.src}}"
+          destination.mac: "{{parsed_event.message.dstmac}}"
+          destination.port: "{{parsed_event.message.dstport or parsed_event.message.dpt}}"
+          destination.domain: "{{parsed_event.message.hostname}}"
+          log.level: "{{parsed_event.message.level or parsed_event.message.FTNTFGTlevel or parsed_event.message.FortinetFortiGatelevel}}"
+          log.description: "{{parsed_event.message.logdesc}}"
+          event.code: "{{parsed_event.message.logid or parsed_event.message.FTNTFGTlogid or parsed_event.message.FortinetFortiGatelogid}}"
+          event.type: "{{parsed_event.message.desc}}"
+          event.reason: "{{parsed_event.message.reason}}"
+          event.message: "{{parsed_event.message.msg}}"
```
Comment (PierrickV) [inline Fortinet/fortigate/ingest/parser.yml:43]: `event.message` is already used to store the original log
```suggestion
          event.reason: "{{parsed_event.message.msg}}"
```

File: Fortinet/fortigate/ingest/parser.yml [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+          source.ip: "{{parsed_event.message.remip or parsed_event.message.srcip or parsed_event.message.src}}"
+          source.address: "{{parsed_event.message.remip or parsed_event.message.srcip or parsed_event.message.src}}"
+          destination.mac: "{{parsed_event.message.dstmac}}"
+          destination.port: "{{parsed_event.message.dstport or parsed_event.message.dpt}}"
+          destination.domain: "{{parsed_event.message.hostname}}"
+          log.level: "{{parsed_event.message.level or parsed_event.message.FTNTFGTlevel or parsed_event.message.FortinetFortiGatelevel}}"
+          log.description: "{{parsed_event.message.logdesc}}"
+          event.code: "{{parsed_event.message.logid or parsed_event.message.FTNTFGTlogid or parsed_event.message.FortinetFortiGatelogid}}"
+          event.type: "{{parsed_event.message.desc}}"
+          event.reason: "{{parsed_event.message.reason}}"
```
Comment (PierrickV) [inline Fortinet/fortigate/ingest/parser.yml:42]:
```suggestion
          event.action: "{{parsed_event.message.reason}}"
```

This field looks more like [event.action](https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-action) to me

File: Fortinet/fortigate/ingest/parser.yml [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+          tmp.icmp_code: "{{parsed_event.message.icmpcode}}"
+          source.user.name: "{{parsed_event.message.user}}"
+          observer.ingress.interface.name: "{{parsed_event.message.srcintf}}"
+          observer.egress.interface.name: "{{parsed_event.message.dstintf}}"
+          observer.type: "{{parsed_event.message.DeviceProduct}}"
+          observer.vendor: "{{parsed_event.message.DeviceVendor}}"
+          observer.version: "{{parsed_event.message.DeviceVersion}}"
+          fortigate.event.severity: "{{parsed_event.message.severity}}"
+          user_agent.original: "{{parsed_event.message.requestClientApplication}}"
+          file.name: "{{parsed_event.message.fname}}"
```
[the remark this answers is not in the document]
Comment (yosradirgham) [reply Fortinet/fortigate/ingest/parser.yml:79]: Thanks for the trick ^^ @PierrickV

File: Fortinet/fortigate/ingest/parser.yml [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+          destination.packets: "{{parsed_event.message.rcvdpkt or parsed_event.message.FTNTFGTrcvpkt or parsed_event.message.FortinetFortiGatercvdpkt or parsed_event.message.get('Packets Received')}}"
+          destination.port: "{{parsed_event.message.dstport or parsed_event.message.dpt}}"
+          dns.question.name: "{{parsed_event.message.qname or parsed_event.message.FTNFGTqname or parsed_event.message.FortinetFortiGateqname}}"
+          dns.question.type: "{{parsed_event.message.qtype or parsed_event.message.FTNFGTqtype or parsed_event.message.FortinetFortiGateqtype}}"
+          event.action: "{{parsed_event.message.reason}}"
+          event.category: "{{parsed_event.message.type}}"
+          event.code: "{{parsed_event.message.logid or parsed_event.message.FTNTFGTlogid or parsed_event.message.FortinetFortiGatelogid}}"
+          event.reason: "{{parsed_event.message.msg}}"
+          event.severity: "{{parsed_event.message.severity}}"
+          event.type: "{{parsed_event.message.desc}}"
```
Comment (PierrickV) [inline Fortinet/fortigate/ingest/parser.yml:50]: `event.type` has a restricted [list of value](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html). Another field should be used

File: Fortinet/fortigate/ingest/parser.yml [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+
+  - name: field_extraction
+
+stages:
+  field_extraction:
+    actions:
+      - set:
+          action.name: "{{parsed_event.message.name or parsed_event.message.FTNTFGTaction or parsed_event.message.FortinetFortiGateaction or parsed_event.message.act or parsed_event.message.action}}"
+          action.outcome: "{{parsed_event.message.status or 'success'}}"
+          action.outcome_reason: "{{parsed_event.message.outcome_reason or parsed_event.message.msg.split('\n')[0].strip('\"')}}"
```
Comment (PierrickV) [inline Fortinet/fortigate/ingest/parser.yml:32]:
```suggestion
          action.outcome_reason: "{{parsed_event.message.outcome_reason or parsed_event.message.msg.split('\n')[0].strip('\"')}}"
```

replaced by `event.reason`

File: Fortinet/fortigate/_meta/fields.yml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+icmp.request.code:
+  name: icmp.request.code
+  type: keyword
+  description: ""
+
+rule.apprisk:
+  name: rule.apprisk
+  type: keyword
+  description: ""
```
Comment (PierrickV) [inline Fortinet/fortigate/_meta/fields.yml:14]:
```suggestion
fortinet.apprisk:
  fortinet: rule.apprisk
  type: keyword
  description: "Risk level of the application."
```

According to this [documentation](https://docs.fortinet.com/document/fortigate/6.2.3/fortios-log-message-reference/357866/log-message-fields) this is the `Risk level of the application.` which is not an ECS field and not a `rule.` field

File: Fortinet/fortigate/tests/WAD.STANDARD.json [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+      "category": "event",
+      "code": "0105048039",
+      "reason": "SSL Alert sent",
+      "created": "2021-04-26T04:17:39.492286Z",
+      "original": "CO0Bc1MbjbOwVYPYTY69yQWWHIKOI8wbccbfhTNgD35U8qHRdBl0E8x0SYR7PAFp",
+      "dialect_uuid": "a948028d-8e9c-48e3-9f3a-193f2c42e9d1",
+      "id": "3b43476c-40d8-45de-91a9-4ee0312bec21",
+      "outcome": "success",
+      "dialect": "fortigate",
+      "type": "illegal parameter",
```
Comment (PierrickV) [inline Fortinet/fortigate/tests/WAD.STANDARD.json:32]: not a valid `event.type`

**Review by PierrickV — APPROVED** [review]
:rocket: 

Thank you for this work

**Pull request merged**